src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -525,84 +602,109 @@
   "Query for the reports entity"
```

Comment:
There may be a similar issue here like the one in the inventory-query with the :producer table not showing up in any of the join deps and getting dropped.